src/main/kotlin/no/nav/medlemskap/services/aareg/AaRegClient.kt

```diff
@@ -0,0 +1,36 @@
+package no.nav.medlemskap.services.aareg
+
+import io.ktor.client.request.get
+import io.ktor.client.request.header
+import io.ktor.client.request.parameter
+import io.ktor.client.request.url
+import io.ktor.http.ContentType
+import io.ktor.http.HttpHeaders
+import no.nav.medlemskap.common.defaultHttpClient
+import no.nav.medlemskap.services.sts.StsRestClient
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+
+
+class AaRegClient(val baseUrl: String, val stsClient: StsRestClient, val callIdGenerator: () -> String) {
+
+    suspend fun hentArbeidsforhold(fnr: String, fraOgMed: LocalDate? = null, tilOgMed: LocalDate? = null): List<Arbeidsforhold> {
+        with(stsClient.oidcToken()) {
```

Comment:
Det viser seg at dette ikke virker. Det blir feil `this` lenger ned i koden.

Reply:
Fikset